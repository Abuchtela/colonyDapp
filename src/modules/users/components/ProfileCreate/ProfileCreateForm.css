.content {
  width: 400px;
}

.actionsContainer {
  display: flex;
  justify-content: flex-end;
}

.title {
  margin-top: 180px;
}
