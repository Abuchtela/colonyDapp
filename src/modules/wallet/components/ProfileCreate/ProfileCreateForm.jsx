/* @flow */

import type { FormikProps } from 'formik';

import React from 'react';
import { defineMessages } from 'react-intl';
import * as yup from 'yup';

import styles from './ProfileCreateForm.css';

import type { FormValues } from './types';

import Input from '../../../core/components/Fields/Input';
import Heading from '../../../core/components/Heading';
import Button from '../../../core/components/Button';

const MSG = defineMessages({
  heading: {
    id: 'ProfileCreate.heading',
    defaultMessage: 'Congrats, your wallet is all set up.',
  },
  subHeading: {
    id: 'ProfileCreate.subHeading',
    defaultMessage: `Let's finish your account!`,
  },
  instructionText: {
    id: 'ProfileCreate.instructionText',
    defaultMessage: 'Add a name so people can identify you within Colony.',
  },
  inputLabel: {
    id: 'ProfileCreate.inputLabel',
    defaultMessage: 'Your name',
  },
  buttonAdvance: {
    id: 'ProfileCreate.buttonAdvance',
    defaultMessage: 'Launch App',
  },
  buttonBack: {
    id: 'ProfileCreate.buttonBack',
    defaultMessage: 'Back',
  },
});

type Props = {
  handleBack: () => void,
} & FormikProps<FormValues>;

const ProfileCreateForm = ({ handleBack, isValid }: Props) => (
  <main className={styles.content}>
    <div className={styles.title}>
      <Heading
        text={MSG.heading}
        appearance={{ size: 'medium', margin: 'none', weight: 'thin' }}
      />
    </div>
    <div className={styles.subTitle}>
      <Heading
        text={MSG.subHeading}
        appearance={{ size: 'medium', weight: 'thin' }}
      />
    </div>
    <div className={styles.instructionText}>
      <Heading
        text={MSG.instructionText}
        appearance={{ size: 'small', weight: 'thin' }}
      />
    </div>
    <Input name="profilename" label={MSG.inputLabel} />
    <div className={styles.actionsContainer}>
      <Button
        text={MSG.buttonBack}
        appearance={{ theme: 'secondary', size: 'large' }}
        onClick={handleBack}
      />
      <Button
        appearance={{ theme: 'primary', size: 'large' }}
        text={MSG.buttonAdvance}
        type="submit"
        disabled={!isValid}
      />
    </div>
  </main>
);

export const validationSchema = yup.object({
  profilename: yup.string().required(),
});

export default ProfileCreateForm;
