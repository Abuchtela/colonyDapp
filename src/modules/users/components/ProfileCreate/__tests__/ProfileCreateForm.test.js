/* eslint-env jest */

import React from 'react';
import toJson from 'enzyme-to-json';

import { shallowWithIntl } from 'testutils';

import ProfileCreateForm from '../ProfileCreateForm.jsx';

describe('ProfileCreateForm component', () => {
  test('Renders initial component (snapshot)', () => {
    const wrapper = shallowWithIntl(<ProfileCreateForm />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
