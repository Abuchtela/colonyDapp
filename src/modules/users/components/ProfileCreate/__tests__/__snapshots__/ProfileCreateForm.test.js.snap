// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileCreateForm component Renders initial component (snapshot) 1`] = `
<main>
  <div>
    <InjectIntl(Heading)
      appearance={
        Object {
          "margin": "none",
          "size": "medium",
          "weight": "thin",
        }
      }
      text={
        Object {
          "defaultMessage": "Congrats, your wallet is all set up.",
          "id": "ProfileCreate.heading",
        }
      }
    />
  </div>
  <div>
    <InjectIntl(Heading)
      appearance={
        Object {
          "size": "medium",
          "weight": "thin",
        }
      }
      text={
        Object {
          "defaultMessage": "Let's finish your account!",
          "id": "ProfileCreate.subHeading",
        }
      }
    />
  </div>
  <div>
    <InjectIntl(Heading)
      appearance={
        Object {
          "size": "small",
          "weight": "thin",
        }
      }
      text={
        Object {
          "defaultMessage": "Add a name so people can identify you within Colony.",
          "id": "ProfileCreate.instructionText",
        }
      }
    />
  </div>
  <InjectIntl(Component)
    label={
      Object {
        "defaultMessage": "Your name",
        "id": "ProfileCreate.inputLabel",
      }
    }
    name="profilename"
  />
  <div>
    <InjectIntl(Button)
      appearance={
        Object {
          "size": "large",
          "theme": "secondary",
        }
      }
      text={
        Object {
          "defaultMessage": "Back",
          "id": "ProfileCreate.buttonBack",
        }
      }
    />
    <InjectIntl(Button)
      appearance={
        Object {
          "size": "large",
          "theme": "primary",
        }
      }
      disabled={true}
      text={
        Object {
          "defaultMessage": "Launch App",
          "id": "ProfileCreate.buttonAdvance",
        }
      }
      type="submit"
    />
  </div>
</main>
`;
